tests/test_cases/test_array_simple/test_array_simple.py

```diff
@@ -0,0 +1,79 @@
+"""
+Test getting and setting values of arrays
+"""
+import logging
+
+import cocotb
+from cocotb.binary import BinaryValue
+from cocotb.triggers import Timer
+
+tlog = logging.getLogger("cocotb.test")
+
+def _check_value(tlog, hdl, expected):
+    if hdl.value != expected:
+        raise TestFailure("{2!r}: Expected >{0}< but got >{1}<".format(expected, hdl.value, hdl))
+    else:
+        tlog.info("   Found {0!r} ({1}) with value={2}".format(hdl, hdl._type, hdl.value))
+
+
+@cocotb.test()
+def test_array_handles(dut):
+    """Test getting and setting array values using the handle of the full array.
+    """
+
+    # Set values with '<=' operator
+    dut.array_7_downto_4 <= [0xF0, 0xE0, 0xD0, 0xC0]
+    dut.array_4_to_7     <= [0xB0, 0xA0, 0x90, 0x80]
+    dut.array_3_downto_0 <= [0x70, 0x60, 0x50, 0x40]
+    dut.array_0_to_3     <= [0x30, 0x20, 0x10, 0x00]
+
+    yield Timer(1000)
+
+    _check_value(tlog, dut.array_7_downto_4, [0xF0, 0xE0, 0xD0, 0xC0])
+    _check_value(tlog, dut.array_4_to_7    , [0xB0, 0xA0, 0x90, 0x80])
+    _check_value(tlog, dut.array_3_downto_0, [0x70, 0x60, 0x50, 0x40])
+    _check_value(tlog, dut.array_0_to_3    , [0x30, 0x20, 0x10, 0x00])
+
+    # Set values through HierarchyObject.__setattr__ method on 'dut' handle
+    dut.array_7_downto_4 = [0x00, 0x11, 0x22, 0x33]
+    dut.array_4_to_7     = [0x44, 0x55, 0x66, 0x77]
+    dut.array_3_downto_0 = [0x88, 0x99, 0xAA, 0xBB]
+    dut.array_0_to_3     = [0xCC, 0xDD, 0xEE, 0xFF]
+
+    yield Timer(1000)
+
+    _check_value(tlog, dut.array_7_downto_4, [0x00, 0x11, 0x22, 0x33])
+    _check_value(tlog, dut.array_4_to_7    , [0x44, 0x55, 0x66, 0x77])
+    _check_value(tlog, dut.array_3_downto_0, [0x88, 0x99, 0xAA, 0xBB])
+    _check_value(tlog, dut.array_0_to_3    , [0xCC, 0xDD, 0xEE, 0xFF])
+
+@cocotb.test()
+def test_array_indexes(dut):
+    """Test getting and setting values of array indexes.
+    """
+
+    # Check indices
+    _check_value(tlog, dut.array_7_downto_4[7], 0x00)
+    _check_value(tlog, dut.array_7_downto_4[4], 0x33)
+    _check_value(tlog, dut.array_4_to_7[4]    , 0x44)
+    _check_value(tlog, dut.array_4_to_7[7]    , 0x77)
+    _check_value(tlog, dut.array_3_downto_0[3], 0x88)
+    _check_value(tlog, dut.array_3_downto_0[0], 0xBB)
+    _check_value(tlog, dut.array_0_to_3[0]    , 0xCC)
+    _check_value(tlog, dut.array_0_to_3[3]    , 0xFF)
+
+    _check_value(tlog, dut.array_0_to_3[1]    , 0xDD)
```

Comment:
I don't like that this relies on the previous test having run - tests should be independent, because they can be run separately via environment variables. Can you copy-paste lines 24 - 28 to this test too?